packages/tools/src/tools/segmentation/ThresholdPreviewTool.ts

```diff
@@ -0,0 +1,235 @@
+      outlineOpacity,
+      outlineOpacityInactive,
+    };
+  };
+
+  setConfiguration = (newConfiguration: Record<string, any>): void => {
+    this.configuration.strategySpecificConfiguration.THRESHOLD_INSIDE_CUBE = {
+      threshold: newConfiguration,
+    };
+
+    this.onSetToolEnabled(null);
+  };
+
+  onSetToolEnabled(event): void {
+    const toolGroupId = this.toolGroupId;
+
+    const toolGroup = getToolGroup(toolGroupId);
+
+    if (!toolGroup) {
+      return;
+    }
+
+    // toolGroup Viewports
+    const toolGroupViewports = toolGroup.viewportsInfo.map(
+      ({ renderingEngineId, viewportId }) => {
+        const enabledElement = getEnabledElementByIds(
+          viewportId,
+          renderingEngineId
+        );
+
+        if (enabledElement) {
+          return enabledElement.viewport;
+        }
+      }
+    );
+
+    const viewport = toolGroupViewports[0];
+
+    const actors = viewport.getActors();
+
+    const firstVolumeActorUID = actors[0].uid;
+    const imageVolume = cache.getVolume(firstVolumeActorUID);
+
+    const activeSegmentationRepresentation =
+      activeSegmentation.getActiveSegmentationRepresentation(toolGroupId);
+
+    const { segmentationId, type } = activeSegmentationRepresentation;
+
+    const { representationData } = getSegmentation(segmentationId);
+
+    const { volumeId } = representationData[type] as LabelmapSegmentationData;
+
+    const segmentation = cache.getVolume(volumeId);
+
+    const segmentIndex =
+      segmentIndexController.getActiveSegmentIndex(segmentationId);
+
+    const operationData = {
+      points: [],
+      imageVolume,
+      volume: segmentation,
+      segmentationId,
+      segmentIndex,
+      segmentsLocked: [],
+      strategySpecificConfiguration:
+        this.configuration.strategySpecificConfiguration,
+    };
+
+    this.applyActiveStrategy(null, operationData);
+
+    function _getSegmentationRepresentation(segmentationId, toolGroupId) {
+      const segmentationRepresentations =
+        segmentationState.getSegmentationRepresentations(toolGroupId);
+
+      if (segmentationRepresentations.length === 0) {
+        return;
+      }
+
+      // Todo: this finds the first segmentation representation that matches the segmentationId
+      // If there are two labelmap representations from the same segmentation, this will not work
+      const representation = segmentationRepresentations.find(
+        (representation) => representation.segmentationId === segmentationId
+      );
+
+      return representation;
+    }
+
+    const segmentationRepresentation = _getSegmentationRepresentation(
+      segmentationId,
+      this.toolGroupId
+    );
+
+    const animationLength = 750;
+
+    // Animation for the segmentIndex
+
+    const { fillAlpha } = this.getConfiguration();
+
+    let count = 0;
+    const intervalTime = 16;
+    const numberOfFrames = Math.ceil(animationLength / intervalTime);
+
+    clearInterval(this.highlightIntervalId);
```

Comment:
we have all the logic above inside `packages/tools/src/tools/displayTools/Labelmap/labelmapDisplay.ts` in `render` almost, and we hit that also on each segmentation updated, so I'm thinking maybe we augment that `render` to be able to handle animations like below. My thought would be to add something like `animationType` Enums, and for now we have `easeInOutBell` and then there if that is presented inside the `segmentationRepresentation` as key, we add animation. What do you think?